### PR TITLE
refactor: (ecam) Convert Cruise page to React

### PR DESCRIPTION
**Pull request by RichardPilbery**
Fixes #[issue_no]

## Summary of Changes

Conversion of Cruise page to React

## Screenshots (if necessary)

ECAM-CRUISE-2
ECAM-CRUISE-1

## Additional context

Discord username (if different from GitHub):

## Testing instructions

1. Programme a flight with a destination
2. Fly it
3. Once airbourne check the following on the  Cruise page:
     a. Check LDG ELEV is auto and the elevation shown matches your destination elevation to the nearest 50ft
     b. Adjust the landing elevation dial on the overhead panel and check elevation changes
     c. Press the cabin pressure mod select button to show MAN and check that arrow is replaced by gauge
     d. Check correct arrow (up or down) is shown when cabin pressurisation is increasing/decreasing 
     e. Adjust the overhead cockpint, forward and aft cabin temperature dials and the temperatures in the bottom left of the screen should start to change (may take 10-30 seconds)

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by MisterChocker** [comment]
I am inexperienced with react, so I am unable to review your pr.

File: flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_LocalVarUpdater.js [hunk 1]
```diff
@@ -211,5 +217,27 @@ class A32NX_LocalVarUpdater {
         );
     }
 
+    _deltaPSI() {
+
```
Comment (Benjozork) [inline flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_LocalVarUpdater.js:221]:
```suggestion
```

File: src/instruments/src/SD/Pages/Crz/Crz.tsx [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+                {rightVN1[0]}
+                .
+            </text>
+            <text id="VibN2RightDecimal" className="Standard Green" x="440" y="265" textAnchor="start">{rightVN1[1]}</text>
+        </>
+    );
+};
+
+export const PressureComponent = () => {
+    const splitDecimals = function (value, type) {
```
Comment (Benjozork) [inline src/instruments/src/SD/Pages/Crz/Crz.tsx:135]: Use an arrow function

File: src/instruments/src/SD/Pages/Crz/Crz.tsx [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+
+export const OilComponent = () => {
+    const splitDecimals = function (value, type) {
+        if (type === 'oil') {
+            value = value * 0.01 * 25;
+        } else if (type === 'vib') {
+            value = value < 0 ? 0.0 : value;
+        }
+        const decimalSplit = value.toFixed(1).split('.', 2);
+        return (decimalSplit);
```
Comment (Benjozork) [inline src/instruments/src/SD/Pages/Crz/Crz.tsx:64]:
```suggestion
        return value.toFixed(1).split('.', 2);
```

File: src/instruments/src/SD/Pages/Crz/Crz.tsx [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+
+            <g
+                id="vsArrow"
+                className={(cabinVs * 60 <= -50 || cabinVs * 60 >= 50) && !manMode ? '' : 'Hide'}
+                transform={cabinVs * 60 <= -50 ? 'translate(0, 795) scale(1, -1)' : 'scale(1, 1)'}
+            >
+                <path d="M433,405 h7 L446,395" className="VsIndicator" strokeLinejoin="miter" />
+                <polygon points="452,388 447,396 457,396" transform="rotate(38,452,388)" className="VsIndicator" />
+            </g>
+
```
Comment (Benjozork) [inline src/instruments/src/SD/Pages/Crz/Crz.tsx:209]:
```suggestion
```

File: src/instruments/src/SD/Pages/Crz/Crz.tsx [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+};
+
+type GaugeComponentType = {
+    x: number,
+    y: number,
+    radius: number,
+    startAngle: number,
+    endAngle: number,
+    verticalSpeed: number,
+    className: string
```
Comment (Benjozork) [inline src/instruments/src/SD/Pages/Crz/Crz.tsx:221]:
```suggestion
    className: string,
```

File: src/instruments/src/SD/Pages/Crz/Crz.tsx [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+    radius: number,
+    startAngle: number,
+    endAngle: number,
+    className: string,
+    showValue: boolean,
+    indicator: boolean
+};
+
+const GaugeMarkerComponent = ({ value, x, y, min, max, radius, startAngle, endAngle, className, showValue, indicator } : GaugeMarkerComponentType) => {
+    const valueRadianAngleConverter = function (value, min, max, endAngle, startAngle) {
```
Comment (Benjozork) [inline src/instruments/src/SD/Pages/Crz/Crz.tsx:268]: Use an arrow function

Comment (Benjozork) [reply src/instruments/src/SD/Pages/Crz/Crz.tsx:268]: Also, you should extract this function to outside the component.

File: src/instruments/src/SD/Pages/Crz/Crz.tsx [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+    let end = {
+        x: x + (dir.x * radius),
+        y: y + (dir.y * radius),
+    };
+
+    if (indicator) {
+        start = {
+            x,
+            y,
+        };
```
Comment (Benjozork) [inline src/instruments/src/SD/Pages/Crz/Crz.tsx:295]:
```suggestion
        start = { x, y };
```

**Review by Benjozork — CHANGES_REQUESTED** [review]

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js [hunk 8]
```diff
@@ -45,6 +45,7 @@ class CDUFlightPlanPage {
             approachRunway = mcdu.flightPlanManager.getApproachRunway();
             if (approachRunway) {
                 destCell += Avionics.Utils.formatRunway(approachRunway.designation);
+                SimVar.SetSimVarValue('L:A32NX_DCDU_APPROACH_RUNWAY_ELEVATION', 'Meters', approachRunway.elevation);
```
Comment (Benjozork) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js:48]: How is this related to the DCDU ?

File: flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_LocalVarUpdater.js [hunk 9]
```diff
@@ -83,6 +83,12 @@ class A32NX_LocalVarUpdater {
                 type: "Bool",
                 selector: this._areSlidesArmed.bind(this),
                 refreshInterval: 100,
+            },
+            {
```
Comment (Benjozork) [inline flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_LocalVarUpdater.js:87]: Please document this new variable in `docs/a320-simvars.md`.

File: src/instruments/src/SD/Pages/Crz/Crz.tsx [hunk 10]
```diff
@@ -0,0 +1,297 @@
...
+    max: number,
+    radius: number,
+    startAngle: number,
+    endAngle: number,
+    className: string,
+    showValue: boolean,
+    indicator: boolean
+};
+
+const GaugeMarkerComponent = ({ value, x, y, min, max, radius, startAngle, endAngle, className, showValue, indicator } : GaugeMarkerComponentType) => {
```
Comment (Benjozork) [inline src/instruments/src/SD/Pages/Crz/Crz.tsx:238]: Can be memoized.

File: src/instruments/src/SD/Pages/Crz/Crz.tsx [hunk 11]
```diff
@@ -0,0 +1,297 @@
...
+    x: number,
+    y: number,
+    min: number,
+    max: number,
+    radius: number,
+    startAngle: number,
+    endAngle: number,
+    className: string,
+    showValue: boolean,
+    indicator: boolean
```
Comment (Benjozork) [inline src/instruments/src/SD/Pages/Crz/Crz.tsx:235]:
```suggestion
    indicator: boolean,
```

File: src/instruments/src/SD/Pages/Crz/Crz.tsx [hunk 12]
```diff
@@ -0,0 +1,297 @@
...
+    x: number,
+    y: number,
+    radius: number,
+    startAngle: number,
+    endAngle: number,
+    verticalSpeed: number,
+    className: string,
+}
+
+const GaugeComponent = ({ x, y, radius, startAngle, endAngle, verticalSpeed, className } : GaugeComponentType) => {
```
Comment (Benjozork) [inline src/instruments/src/SD/Pages/Crz/Crz.tsx:203]: Can be memoized.

File: src/instruments/src/SD/Pages/Crz/common.tsx [hunk 13]
```diff
@@ -0,0 +1,27 @@
+export const splitDecimals = (value, type) => {
```
Comment (Benjozork) [inline src/instruments/src/SD/Pages/Crz/common.tsx:1]: This file is called `common`, but what is it common to ?

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js (hunk 8 above)
Comment (RichardPilbery) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js:48]: Nothing really, Can't remember why I called it that. I'll revise

Comment (MisterChocker) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js:48]: Question, why do we need to write to this simvar over and over again with nothing actually changing?

Why don't we change the simvar when we actually change the runway?

File: src/instruments/src/SD/Pages/Crz/common.tsx (hunk 13 above)
Comment (RichardPilbery) [reply src/instruments/src/SD/Pages/Crz/common.tsx:1]: Just common to the cruise page at the moment, but I might use some of these functions again in which case they could be moved to SD > Common. Or I can move it now?

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js (hunk 8 above)
Comment (RichardPilbery) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js]: Sounds good. Can you suggest the file and code block it needs to be moved to?

File: src/instruments/src/SD/Pages/Crz/Crz.tsx [hunk 14]
```diff
@@ -0,0 +1,301 @@
...
+
+            <text className="SubTitle" x="50" y="60">ENG</text>
+            <FuelComponent />
+            <OilComponent />
+
+            <text className="SubTitle" x="50" y="330">AIR</text>
+            <PressureComponent />
+
+            <CondComponent />
+
```
Comment (Benjozork) [inline src/instruments/src/SD/Pages/Crz/Crz.tsx:25]:
```suggestion
```

File: src/instruments/src/SD/Pages/Crz/Crz.tsx [hunk 15]
```diff
@@ -0,0 +1,301 @@
...
+        </svg>
+    </>
+);
+
+export const FuelComponent = () => {
+    const unit = parseFloat(NXDataStore.get('CONFIG_USING_METRIC_UNIT', '1'));
```
Comment (Benjozork) [inline src/instruments/src/SD/Pages/Crz/Crz.tsx:31]: You can use the `usePersistentProperty` hook here:

https://github.com/flybywiresim/a32nx/blob/bbb804a8bd43180830fd97b3996ef1df48785fe8/src/instruments/src/Common/persistence.tsx#L41

**Review by Benjozork — CHANGES_REQUESTED** [review]
Very little things remaining, then we are good.

File: src/instruments/src/SD/Pages/Crz/Crz.tsx (hunk 15 above)
Comment (RichardPilbery) [reply src/instruments/src/SD/Pages/Crz/Crz.tsx:30]: Done

**Review by Benjozork — APPROVED** [review]

**Comment by marcman86** [comment]
Quality Assurance Tester Report
Discord:marcman86#4907
Object of testing: #5086
Tier of Testing: 1
Date: 10/07/2021

Testing Process:
- Spawned on runway at EDDP
- Typed in missing values in MCDU (FL, CI,...) and took off
- Checked CRUISE page and its values
- Set CABIN PRESS to manual

Negatives: -

Testing Results: Passed
Conclusions: Works
Hope this helps :-)

**Comment by St54Kevin** [comment]
**Quality Assurance Tester**

Object of testing: #5086 
Tier of Testing: 2
Date: 12/07/2021

**Testing Process:**
1. Full VATSIM flight EDDF-EDDB
2. Checked LDG ELEV modes
3. Checked MAN press modes
4. Checked temps

**Negatives:** N/A
**Testing Results:** Passed
